### PR TITLE
Fix ColorSwatch 1px shift when isWhite swatch is present at the start of or within a row (not the end)

### LINKED ISSUE #6205: On Hover issue in swatch color picker
### Bug Report 

- __Package version(s)__: 6.55.0
- __Browser and OS versions__: (fill this out if relevant)
 
#### Priorities and help requested:
 
Are you willing to submit a PR to fix? 
 
Requested priority: Normal
 
Products/sites affected: (if applicable)
 
#### Describe the issue:
For Swatch Color picker if there is a white color cell and that cell is not at the rightmost end -  then if we hover over on the white cell , all the other colors(on the right of the white ) move 1-2 px to the left .

#### Actual behavior:
In our scenario , we have a white cell at the first(left most) position in a swatch color picker , Due to this issue all the color cells on its right(columns/rows) are moving to 1-2 px to left on hover .
 
#### Expected behavior:
Color cells should not move on hover 

Sample GIF :
bug

### PR DESCRIPTION
**Pull request by KevinTCoughlin**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #6205
- [x] Include a change request file using `$ npm run change`

#### Description of changes

**Bugs**

TLDR: See [GIFs](https://github.com/OfficeDev/office-ui-fabric-react/pull/6222#issuecomment-420784113)

- The color swatches in `SwatchColorPicker` shift `1px` with `:focus` regardless of where the `white` swatch resides.
- The color swatches shift `1px` with `:hover` when a `white` swatch is placed within a row of swatches and not the start or end.

These bugs occur because we are setting a border on the `white` swatch so it is discernible against the same color background. However, we change the border on `:hover` and `:focus` to be larger (`4px`), losing the `1px` border around the `white` SVG.

**The Fix**

We need to maintain that internal `1px` "border" so that the size isn't `1px` smaller for the `white` swatch when hovered or focused. But we also need the `4px` "gray" border on `:hover` and `:focus`. Since we can't have two borders at once, I'm using an inset box-shadow and toggling `transparent` between it and the `border-color` depending on state. This ensures we don't lose dimensions when transitioning states leading to row shift.

That was a fun one 😄 .

#### Focus areas to test

On `master`, change an example's swatch array to have `#ffffff` anywhere but the end of a row and you'll see the 1px shift on `master`. Here's a diff that repros with it at the start of the row:

```diff
--- a/packages/office-ui-fabric-react/src/components/SwatchColorPicker/examples/SwatchColorPicker.Basic.Example.tsx
+++ b/packages/office-ui-fabric-react/src/components/SwatchColorPicker/examples/SwatchColorPicker.Basic.Example.tsx
@@ -27,11 +27,11 @@ export class SwatchColorPickerBasicExample extends React.Component<any, IBasicSw
           columnCount={5}
           cellShape={'circle'}
           colorCells={[
             { id: 'a', label: 'green', color: '#00ff00' },
+            { id: 'e', label: 'white', color: '#ffffff' },
             { id: 'b', label: 'orange', color: '#ffa500' },
             { id: 'c', label: 'blue', color: '#0000ff' },
-            { id: 'd', label: 'red', color: '#ff0000' },
```

### CODE REVIEW

**Comment by micahgodbolt** [comment]
can you toss in a gif of the new experience?

**Comment by KevinTCoughlin** [comment]
While recording a GIF I noticed the following states also have this issue:

- Selected
- Focus (via Tab and LEFT/RIGHT)

Closing for a few, will re-open with those fixes.

**Comment by Jahnp** [comment]
Clever solution @KevinTCoughlin !
